### PR TITLE
#11 Move base classes to separate module

### LINKED ISSUE #11: Move base classes to separate module
Separate base classes (is it worth being called a framework?) into own module and have Swapi as an example application

### PR DESCRIPTION
**Pull request by jonyas**
Close #11

### CODE REVIEW

File: app/src/main/java/anstaendig/com/architecturecomponents/injection/AppInjector.kt [hunk 1]
```diff
@@ -7,7 +7,7 @@ import android.support.v4.app.Fragment
 import android.support.v4.app.FragmentActivity
 import android.support.v4.app.FragmentManager
 import anstaendig.com.architecturecomponents.App
-import anstaendig.com.architecturecomponents.ui.base.BaseActivity
+import com.anstaendig.base.ui.BaseActivity
```
Comment (anstaendig) [inline app/src/main/java/anstaendig/com/architecturecomponents/injection/AppInjector.kt]: Should it not be `com.anstaendig.architecturecomponents.base`?

**Comment by anstaendig** [comment]
I'm not sure though if `ViewModelKey` should really be in the base module as all the injection related classes are not?

File: app/src/main/java/anstaendig/com/architecturecomponents/injection/AppInjector.kt (hunk 1 above)
Comment (jonyas) [inline app/src/main/java/anstaendig/com/architecturecomponents/injection/AppInjector.kt]: this is UI right? Then I think it's right

**Comment by jonyas** [comment]
Agree on the ViewModelKey, I'll move it tomorrow and add it to this PR

**Pull request merged**